### PR TITLE
rExtract types

### PR DESCRIPTION
**Pull request by olafurpg**

### CODE REVIEW

File: plugin/src/main/scala/org/scalameta/paradise/converters/PersistPhase.scala [hunk 1]
```diff
@@ -47,3 +53,26 @@ trait PersistPhase extends ReflectToolkit with Converter {
...
+      override def run(): Unit = {
+        println("HELLO FROM COMPILER!")
+        global.currentRun.units.foreach(unit => {
+          val fixedTree = unit.fix
+          Files.write(Paths.get(unit.source.path), fixedTree.getBytes("UTF-8"))
+        })
+      }
+    }
+  }
+}
```
Comment (xeno-by) [inline plugin/src/main/scala/org/scalameta/paradise/converters/PersistPhase.scala]: Why not have this as a separate class in a separate package? It's not a converter, right?

File: plugin/src/main/scala/org/scalameta/paradise/Plugin.scala [hunk 2]
```diff
@@ -2,27 +2,32 @@ package org.scalameta.paradise
...
     with AnalyzerPlugins
     with HijackBackend {
   val name        = "macroparadise"
   val description = "Empowers production Scala compiler with latest macro developments"
   val components = {
-    val persistEnabled = sys.props("persist.enable") != null
+    val persistEnabled  = sys.props("persist.enable") != null
+    val scalafixEnabled = sys.props("scalafix.enable") != null
     if (persistEnabled) List[NscPluginComponent](PersistComponent)
+    else if (scalafixEnabled || true) List[NscPluginComponent](ScalafixComponent)
```
Comment (xeno-by) [inline plugin/src/main/scala/org/scalameta/paradise/Plugin.scala]: `|| true`? :)

File: tests/converter/src/test/scala/org/scalameta/tests/ConverterSuite.scala [hunk 3]
```diff
@@ -83,15 +90,15 @@ trait ConverterSuite extends FunSuiteLike {
...
                   loop(xctor, yctor)
-                case (ctor"$xctor(...${ Seq(Seq()) })", ctor"$yctor(...${ Seq() })") =>
+                case (ctor"$xctor(...${Seq(Seq())})", ctor"$yctor(...${Seq()})") =>
                   loop(xctor, yctor)
                 case (p"$xpat @ _", p"$ypat") =>
                   loop(xpat, ypat)
                 case (p"$xlhs @ (_: $xtpe)", p"$ylhs: $ytpe") =>
                   loop(xlhs, ylhs) && loop(xtpe, ytpe)
-                case (t"${ Some(xtpe) } {}", t"$ytpe") =>
+                case (t"${Some(xtpe)} {}", t"$ytpe") =>
```
Comment (xeno-by) [inline tests/converter/src/test/scala/org/scalameta/tests/ConverterSuite.scala]: It looks like scalafmt oscillates between these coding styles in different releases :)

File: plugin/src/main/scala/org/scalameta/paradise/Plugin.scala (hunk 2 above)
Comment (olafurpg) [inline plugin/src/main/scala/org/scalameta/paradise/Plugin.scala]: Was debugging :p

File: plugin/src/main/scala/org/scalameta/paradise/converters/PersistPhase.scala (hunk 1 above)
Comment (olafurpg) [inline plugin/src/main/scala/org/scalameta/paradise/converters/PersistPhase.scala]: I was just experimenting, the final destination of the plugin will be in the scalafix repo.

File: tests/converter/src/test/scala/org/scalameta/tests/ConverterSuite.scala (hunk 3 above)
Comment (olafurpg) [inline tests/converter/src/test/scala/org/scalameta/tests/ConverterSuite.scala]: This was fixed in 0.4.8, there should be no space.

**Comment by olafurpg** [comment]
This was just an experiment not intended to be merged.

**Pull request closed without merging**